src/auto-complete.jsx

```diff
@@ -349,8 +355,9 @@ const AutoComplete = React.createClass({
     };
 
     const requestsList = [];
+    const maxSearchResults = this.props.maxSearchResults;
 
-    this.props.dataSource.map((item) => {
+    for (let item of this.props.dataSource) {
```

Comment:
Could we not use `for of` ? (https://github.com/callemall/material-ui/pull/2904) This require the use of babel-polyfill for the `Symbol`